capsules/src/analog_sensor.rs

```diff
@@ -0,0 +1,118 @@
+//! Capsule for analog sensors.
+//!
+//! This capsule provides the sensor HIL interfaces for sensors which only need
+//! an ADC.
+//!
+//! It includes support for analog light sensors and analog temperature sensors.
+
+use kernel::common::cells::OptionalCell;
+use kernel::hil;
+use kernel::ReturnCode;
+
+/// The type of the sensor implies how the raw ADC reading should be converted
+/// to a light value.
+pub enum AnalogLightSensorType {
+    LightDependentResistor,
+}
+
+pub struct AnalogLightSensor<'a, A: hil::adc::Adc> {
+    adc: &'a A,
+    channel: &'a <A as hil::adc::Adc>::Channel,
+    sensor_type: AnalogLightSensorType,
+    client: OptionalCell<&'a hil::sensors::AmbientLightClient>,
+}
+
+impl<A: hil::adc::Adc> AnalogLightSensor<'a, A> {
+    pub fn new(
+        adc: &'a A,
+        channel: &'a <A as hil::adc::Adc>::Channel,
+        sensor_type: AnalogLightSensorType,
+    ) -> AnalogLightSensor<'a, A> {
+        AnalogLightSensor {
+            adc: adc,
+            channel: channel,
+            sensor_type: sensor_type,
+            client: OptionalCell::empty(),
+        }
+    }
+}
+
+/// Callbacks from the ADC driver
+impl<A: hil::adc::Adc> hil::adc::Client for AnalogLightSensor<'a, A> {
+    fn sample_ready(&self, sample: u16) {
+        // TODO: calculate the actual light reading.
```

Comment:
Are these `TODO`s the things that are blocking on the ADC HIL PR?

Reply:
They were, but now they are blocking on testing.